src/gui/log/logmodel.cpp

```diff
@@ -118,7 +118,7 @@ void BaseLogModel::addNewMessage(const BaseLogModel::Message &message)
     // if row is inserted on filled up buffer, the size will not change
     // but because of calling of beginInsertRows function we'll have ghost rows.
     if (m_messages.size() == MAX_VISIBLE_MESSAGES) {
-        const int lastMessage = m_messages.size() - 1;
+        const int lastMessage = MAX_VISIBLE_MESSAGES - 1;
```

Comment:
Read the code again, please.

Reply:
I see it. Your code is still less maintainable. In old one you can easily change the condition of removing last message but body keeps valid (since `lastMessage` points to last message independently from condition above). Your code require body to be changed as well.

Reply:
Sorry, but are you trying to prove something? What maintainability? We are talking about two consecutive lines of code. And actually the `MAX_VISIBLE_MESSAGES` value is "logic". It is a limit for number of messages stored and this very block of code is guarding that. Changing this condition to something else, while leaving body untouch is not something that would ever make sense.

Reply:
Just apply `static_cast` like in other similar places.